Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -77,6 +78,8 @@ AuxiliaryProcessProxy::AuxiliaryProcessProxy(bool alwaysRunsAtBackgroundPriority
 
 AuxiliaryProcessProxy::~AuxiliaryProcessProxy()
 {
+    m_throttler.invalidateAllActivitiesAndDropAssertion();
```

Comment:
I'm still not convinced this is safe, as I assume this can end up calling virtual functions on the AuxiliaryProcessProxy. Maybe we should set a flag in ~ProcessThrottler() to keep track of the fact that we started destruction and not call the bad virtual function in this case? I assume the one that's causing trouble is calling prepareToSuspend() on destruction when releasing the assertion?

Reply:
Ah, good point! As you say, this is also not correct, since we can still be calling virtual methods in AuxiliaryProcessProxy from ~AuxiliaryProcessProxy. The issue is that `ProcessThrottler::sendPrepareToSuspendIPC`, which has several AuxiliaryProcessProxy virtual method calls can be called from ~AuxiliaryProcessProxy. As you suggest, a flag is a good approach to solve this. I am also looking into a patch which calls ProcessThrottler::didDisconnectFromProcess from AuxiliaryProcessProxy::shutDownProcess.

Thanks for reviewing!